src/line_launcher/mod.rs

```diff
@@ -29,20 +31,29 @@ impl LineLauncher {
     ) {
         let mut midi_message_sender = MidiMessageSender { output };
         let mut state = State::NotPlaying;
+        let mut has_gotten_first_downbeat_message = false;
```

Comment:
This is just to be able to initialize `chord_index` to `0` (since you can't set a `usize` to `-1`) but not increment it to `1` on the first downbeat

Reply:
That smells like state that could be represented with an enum.

Reply:
Actually, why is that not included in the main `State` enum?

Reply:
Ok it seems like a mismatch for the existing `State` enum because that's representing "playing state" (which in the bigger picture doesn't necessarily have anything to do with how many downbeats we've seen) whereas this is sort of global "how much time has passed" state - but then ya it makes sense how this + `chord_index` could be represented as a single enum representing "chord progression state"

So updated, I ended up with a `ProgressionState` struct + a nested `ProgressionChordIndexState` enum, would be curious for you to take a look at that modeling